crates/engine/tree/src/tree/payload_validator.rs

```diff
@@ -343,6 +358,20 @@ where
             };
         }
 
+        /// A helper macro for handling errors after the input has been converted to a block
```

Comment:
this was moved up and renamed. not sure if we should keep the macro here but felt that it made sense without having to create helper functions

Reply:
can probably unify them into one `macro_rules` declaration that has two variants: with one and two expressions